src/com/amazon/corretto/crypto/provider/AccessibleByteArrayOutputStream.java

```diff
@@ -83,11 +87,22 @@ byte[] getDataBuffer() {
   }
 
   void reset() {
-    Arrays.fill(buf, 0, count, (byte) 0);
+    int sizeUsed = count;
+    Arrays.fill(buf, 0, sizeUsed, (byte) 0);
```

Comment:
This I/O work is wasted in the shrinkage case.

Reply:
It looks like you zeroize the original buffer when growing, too, even though it is not reused, so I assume doing the same when shrinking is desirable.